### ISSUE TITLE
When a crash occurs, the error message is printed twice

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8964]
**Tell us about your environment**

* **ESLint Version:** 4.2.0
* **Node Version:** 8.1.4
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
rules:
  no-restricted-syntax: [error, 'Invalid selector [[[[']
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
a;
```

**What did you expect to happen?**

The configuration isn't valid (it includes an invalid selector), so I expected ESLint to output an error message.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint outputted the same error message twice, once with the stack trace and once without:

<details>
<summary>Output</summary>

```
Syntax error in selector "Invalid selector [[[[" at position 18: Expected " ", "." or [^ [\],():#!=><~+.] but "[" found.
Error: Syntax error in selector "Invalid selector [[[[" at position 18: Expected " ", "." or [^ [\],():#!=><~+.] but "[" found.
    at tryParseSelector (/path/to/node_modules/eslint/lib/util/node-event-generator.js:163:19)
    at lodash.memoize.rawSelector (/path/to/node_modules/eslint/lib/util/node-event-generator.js:175:28)
    at memoized (/path/to/node_modules/lodash/lodash.js:10566:27)
    at NodeEventGenerator.eventNames.forEach.rawSelector (/path/to/node_modules/eslint/lib/util/node-event-generator.js:235:30)
    at Array.forEach (native)
    at new NodeEventGenerator (/path/to/node_modules/eslint/lib/util/node-event-generator.js:234:20)
    at Linter.verify (/path/to/node_modules/eslint/lib/linter.js:909:34)
    at processText (/path/to/node_modules/eslint/lib/cli-engine.js:203:31)
    at processFile (/path/to/node_modules/eslint/lib/cli-engine.js:245:18)
    at executeOnFile (/path/to/node_modules/eslint/lib/cli-engine.js:585:25)

```

</details>

This isn't specific to the `no-restricted-syntax` rule -- it seems to occur whenever any error is thrown from anywhere.

**Comment by not-an-aardvark** [comment]
Reopening because https://github.com/eslint/eslint/commit/a5fd1011bd4bd9ea119d70c1d980816f96672f97 was effectively reverted in https://github.com/eslint/eslint/pull/9041 due to https://github.com/eslint/eslint/issues/9011.

**Comment by nzakas** [comment]
Looks like @VictorHom has a fix on his fork; I'll create a pull request for that.
